### PR TITLE
Simplecov

### PR DESCRIPTION
**Pull request by CheezItMan**
## Description
Added new lesson on SimpleCov.

## Todos
Are any of these items still in progress?
- [ ]  I may want to add an example where SimpleCov exposes code that will never be executed.  

## Feedback Requested
Feedback is welcome.

### CODE REVIEW

**Comment by droberts-ada** [comment]
Did you mean to add that picture of your insurance statement to the repo?

Other than that, two thumbs up!

File: 00-programming-fundamentals/16-code-coverage-and-simplecov.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+
+## So If I have 100% Coverage I'm Good Right?  No Bugs?
+
+Ummm... no.  SimpleCov measures which statements were executed via your tests, not every combination of values possible.  It also can't guarantee your tests are 100 error free as well.  So bugs can creep in.  However a good amount of code coverage does provide you with insights into your testing.  It can:
+
+-  Help you identify code that may be unnecessary.
+-  Help you identify "blind spots" testing.
+
+So code coverage is a tool that's a useful indicator of the test quality, but having high code coverage doesn't guarantee that your testing is perfect.  It can be especially useful if you don't write tests first, as it can be easy to forget to test some functionality and code coverage reports can help you identify areas without testing.  It's also helpful as one indicator of quality in your tests, just don't assume if you have high coverage you'll have perfect code.  Some teams require a certain percentage of code coverage and it's likely you will encounter coverage requirements in your career.
```
Comment (droberts-ada) [inline 00-programming-fundamentals/16-code-coverage-and-simplecov.md]: This paragraph is pretty long - maybe break it up or remove some redundancy?

File: 00-programming-fundamentals/16-code-coverage-and-simplecov.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+Notice that it's telling us we're not testing two of the conditions.  
+
+
+## So If I have 100% Coverage I'm Good Right?  No Bugs?
+
+Ummm... no.  SimpleCov measures which statements were executed via your tests, not every combination of values possible.  It also can't guarantee your tests are 100 error free as well.  So bugs can creep in.  However a good amount of code coverage does provide you with insights into your testing.  It can:
+
+-  Help you identify code that may be unnecessary.
+-  Help you identify "blind spots" testing.
```
Comment (droberts-ada) [inline 00-programming-fundamentals/16-code-coverage-and-simplecov.md]: I think you might be missing a preposition here.

**Comment by kariabancroft** [comment]
Looks good! Thanks Chris!

**Pull request merged**